validate/validate.go

```diff
@@ -490,6 +504,89 @@ func namespaceValid(ns rspec.Namespace) bool {
 	return true
 }
 
+// Check whether pathB is nested whithin pathA
+func nestedValid(os, pathA, pathB string) (bool, error) {
```

Comment:
I didn't read through, according to your comments, isn't this something we can do with just `filepath.Clean()` and `filepath.Rel()`?

And I don't know how you handle with different oses other places in runtime-tools, but why not separate them in different files, distinguish with build tags or different command options. I'll be more extendable and make codes neat and clean.